### PR TITLE
Accessible dialog component v1

### PR DESCRIPTION
**Pull request by siddharthkp**
- [x] added component
- [x] added stories/tests

supporting changes: IconButton supports innerRef now

Pending decisions:
- [ ] should _close when clicked outside_ be optional (thinking comments)
- [ ] fix upstream bug https://github.com/reach/reach-ui/issues/390

Probably not in this PR:
- [ ] add option for animate from trigger position (thinking comments)

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3706/commits/e9f544e174a07cb71ae182e6eab7a148d151e346 failed.

File: packages/components/src/components/Dialog/index.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    dialogVisible,
+    setDialogVisibility,
+    triggerRect,
+    label,
+  } = React.useContext(context);
+
+  const [overlayVisible, setOverlayVisiblity] = React.useState(false);
+
+  React.useEffect(() => {
+    if (dialogVisible) setOverlayVisiblity(true);
```
Comment (MichaelDeBoey) [inline packages/components/src/components/Dialog/index.tsx]: Put curly braces around body to make it less error prone

```suggestion
    if (dialogVisible) {
      setOverlayVisiblity(true);
    }
```

File: packages/components/src/components/Dialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            )}
+          </AnimatePresence>
+        )}
+      </Rect>
+    </DialogOverlay>
+  );
+};
+
+const centered = (triggerRect, dialogRect) => {
+  if (!dialogRect || !triggerRect) return { left: 0, top: 0 };
```
Comment (MichaelDeBoey) [inline packages/components/src/components/Dialog/index.tsx]: Same

```suggestion
  if (!dialogRect || !triggerRect) {
    return { left: 0, top: 0 };
  }
```

File: packages/components/src/components/Dialog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+const context = React.createContext({
+  dialogVisible: false,
+  setDialogVisibility: null,
+  triggerRef: null,
+  triggerRect: null,
+  label: null,
+});
+
+type DialogTypes = React.FC<{
```
Comment (MichaelDeBoey) [inline packages/components/src/components/Dialog/index.tsx]:
```suggestion
type DialogTypes = FunctionComponent<{
```

File: packages/components/src/components/Dialog/index.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
+import React from 'react';
```
Comment (MichaelDeBoey) [inline packages/components/src/components/Dialog/index.tsx]:
```suggestion
import React, { FunctionComponent } from 'react';
```

File: packages/components/src/components/Dialog/index.tsx (hunk 3 above)
Comment (siddharthkp) [reply packages/components/src/components/Dialog/index.tsx]: Not a fan of this, sorry!

**Pull request closed without merging**